verilog/tools/kythe/kythe_facts.cc

```diff
@@ -17,15 +17,45 @@
 #include <iostream>
 #include <string>
 
+#include "absl/strings/escaping.h"
 #include "absl/strings/substitute.h"
 
 namespace verilog {
 namespace kythe {
 
+bool Signature::operator==(const Signature& other) const {
+  return this->ToString() == other.ToString();
+}
+
+bool Signature::operator<(const Signature& other) const {
+  return this->ToString() < other.ToString();
+}
+
+void Signature::AppendName(absl::string_view name) {
+  names.push_back(std::string(name));
+}
+
+bool Signature::IsEqualToIgnoringScope(absl::string_view name) const {
+  return names.back() == name;
+}
+
+std::string Signature::ToString() const {
+  std::string signature = "";
+  for (const std::string name : names) {
```

Comment:
this is creating a copy of a string at every iteration
either make it std::string& or absl::string_view (string_view is preferred)